analyzer/src/main/kotlin/managers/utils/PubCacheReader.kt

```diff
@@ -64,14 +63,26 @@ internal class PubCacheReader {
             .find { !it.isSymbolicLink() && it.isFile && it.name == filename }
     }
 
-    fun findProjectRoot(packageInfo: JsonNode): File? {
+    fun findProjectRoot(packageInfo: JsonNode, workingDir: File = File("")): File? {
         val packageVersion = packageInfo["version"].textValueOrEmpty()
         val type = packageInfo["source"].textValueOrEmpty()
         val description = packageInfo["description"]
         val packageName = description["name"].textValueOrEmpty()
         val url = description["url"].textValueOrEmpty()
         val resolvedRef = description["resolved-ref"].textValueOrEmpty()
+        val resolvedPath = description["path"].textValueOrEmpty()
+        val isPathRelative = description["relative"]
+
+        if (type == "path" && resolvedPath.isNotEmpty()) {
+            // Packages with source set to "path" and a "path" key in description set to a path.
+            // The path should be kept as is
 
+            if ( isPathRelative.toString().toBoolean() == true ) {
+                return workingDir.resolve(resolvedPath).takeIf {it.isDirectory}
+            } else {
+                return java.io.File(resolvedPath).takeIf {it.isDirectory}
```

Comment:
Both `return` statements should be replaced by a single `return` before the `if`.